dgraph/cmd/alpha/admin.go

```diff
@@ -25,54 +25,83 @@ import (
 	"net/http"
 	"strconv"
 
+	"github.com/dgraph-io/dgraph/edgraph"
 	"github.com/dgraph-io/dgraph/posting"
 	"github.com/dgraph-io/dgraph/worker"
 	"github.com/dgraph-io/dgraph/x"
 	"github.com/golang/glog"
 )
 
+func checkIpIsWhitelisted(w http.ResponseWriter, r *http.Request) bool {
+	ip, _, err := net.SplitHostPort(r.RemoteAddr)
+	if err != nil || (!ipInIPWhitelistRanges(ip) && !net.ParseIP(ip).IsLoopback()) {
+		x.SetStatus(w, x.ErrorUnauthorized, fmt.Sprintf("Request from IP: %v", ip))
+		return false
+	}
+	return true
+}
+
+func checkPoormansAcl(w http.ResponseWriter, r *http.Request) bool {
+	if worker.Config.AuthToken != "" && worker.Config.AuthToken != r.Header.Get(
+		"X-Dgraph-AuthToken") {
+		x.SetStatus(w, x.ErrorUnauthorized, "Invalid X-Dgraph-AuthToken")
+		return false
+	}
+	return true
+}
+
 // handlerInit does some standard checks. Returns false if something is wrong.
-func handlerInit(w http.ResponseWriter, r *http.Request, allowedMethods map[string]bool) bool {
+func handlerInit(w http.ResponseWriter, r *http.Request, allowedMethods map[string]bool,
+	allowOnlyGuardians bool) bool {
 	if _, ok := allowedMethods[r.Method]; !ok {
 		x.SetStatus(w, x.ErrorInvalidMethod, "Invalid method")
+		w.WriteHeader(http.StatusMethodNotAllowed)
```

Comment:
Isn't this a bit strange? Middlewares should either do the `return true/false` pattern, or they should `call next() on success`. This feels like it's mixing both